### ISSUE TITLE
neural network workflow: too many channels will cause model to silently fail

### ISSUE THREAD

**Issue by k-dominik** [ilastik/ilastik#2678]
When checking for incompatible shapes, the current code

https://github.com/ilastik/ilastik/blob/e093db405cda4a4eeb529aebebbc6f55f339a5a5/ilastik/applets/neuralNetwork/nnClassGui.py#L767-L770

only checks if data extend is at least the same as the  minimal model shape. This works for the spacial axes `x`, `y`, `z`. However, in the channel axes, this usually doesn't make sense - those are not treated with padding, or tiling to accommodate for larger sizes.

example model: [affable-shark](https://bioimage.io/#/?tags=affable-shark&id=10.5281%2Fzenodo.5764892). There the training data is multi-channel, but for training it was flattened to a single channel. Then trying to predict on the "original" training/testing/validation set silently fails in ilastik :/

**Comment by constantinpape** [comment]
> Then trying to predict on the "original" training/testing/validation set silently fails in ilastik :/

Just as a comment: the original training data is already single channel, see https://github.com/bioimage-io/collection-bioimage-io/issues/540#issuecomment-1438851292.

(But this is for sure a bit confusing for this model, and overall this should throw some more meaningful error.)
